File: android/app/src/main/java/com/turbomodules/legacy/FlatListDataModuleLegacy.kt
package com.turbomodules.legacy

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.WritableNativeArray
import com.facebook.react.bridge.WritableNativeMap
import kotlin.random.Random

/**
 * 🐌 LEGACY BRIDGE - FlatList Data Generator (Android)
 * 
 * Uses React Native Bridge (slower for large data transfers)
 * Data must be serialized to JSON and sent through bridge queue
 */
class FlatListDataModuleLegacy(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    override fun getName() = "FlatListDataModuleLegacy"

    @ReactMethod
    fun generateData(count: Double, promise: Promise) {
        try {
            // Legacy: Goes through bridge (JSON serialization overhead)
            val data = WritableNativeArray()
            val timestamp = System.currentTimeMillis()
            
            for (i in 0 until count.toInt()) {
                val item = WritableNativeMap()
                item.putString("id", "item_$i")
                item.putString("title", "Item $i")
                item.putString("description", "This is the description for item number $i")
                item.putInt("value", Random.nextInt(1000))
                item.putDouble("timestamp", timestamp.toDouble())
                
                data.pushMap(item)
            }
            
            // Data will be serialized to JSON by bridge (slower)
            promise.resolve(data)
        } catch (e: Exception) {
            promise.reject("ERROR", "Failed to generate data", e)
        }
    }
}

File: ios/TurboModules/FlatListDataModuleLegacy.m
/**
 * 🐌 LEGACY BRIDGE IMPLEMENTATION - FlatList Data Generator
 * 
 * Uses React Native Bridge (slower for large data transfers)
 * Data must be serialized to JSON and sent through bridge queue
 */

#import "FlatListDataModuleLegacy.h"

@implementation FlatListDataModuleLegacy

RCT_EXPORT_MODULE()

// 🔑 REQUIRES RCT_EXPORT_METHOD macro
RCT_EXPORT_METHOD(generateData:(double)count
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)
{
  @try {
    NSMutableArray *data = [NSMutableArray arrayWithCapacity:(int)count];
    
    for (int i = 0; i < (int)count; i++) {
      NSDictionary *item = @{
        @"id": [NSString stringWithFormat:@"item_%d", i],
        @"title": [NSString stringWithFormat:@"Item %d", i],
        @"description": [NSString stringWithFormat:@"This is the description for item number %d", i],
        @"value": @(arc4random_uniform(1000)),
        @"timestamp": @([[NSDate date] timeIntervalSince1970] * 1000)
      };
      [data addObject:item];
    }
    
    // Data will be serialized to JSON by bridge (slower)
    resolve(data);
  } @catch (NSException *exception) {
    reject(@"ERROR", @"Failed to generate data", nil);
  }
}

@end
